main.py

```diff
@@ -72,14 +73,27 @@ def process_movie_section(s, watched_set, ratings_dict, listutil, collection):
                 logging.error('Movie [{} ({})]: Not found. Aborting'.format(
                     movie.title, movie.year))
                 continue
-
+            last_time = time()
             if CONFIG['sync']['collection']:
                 # add to collection if necessary
                 if m.trakt not in collection:
-                    logging.info('Movie [{} ({})]: Added to trakt collection'.format(
-                        movie.title, movie.year))
-                    m.add_to_library()
-
+                    retry = 0
+                    while retry < 5:
+                        try:
+                            last_time = respect_trakt_rate(last_time)
+                            m.add_to_library()
+                            logging.info('Movie [{} ({})]: Added to trakt collection'.format(
+                                movie.title, movie.year))
+                            break
+                        except trakt.errors.RateLimitException as e:
+                            delay = int(e.response.headers.get("Retry-After", 1))
+                            logging.warning(
+                                "Movie [{} ({})]: Rate Limited on adding to collection. Sleeping {} sec from trakt (GUID: {})".format(movie.title, movie.year, delay, guid))
+                            sleep(delay)
+                            retry += retry
```

Comment:
what does `retry += retry` supposed to do? seems to be `retry += 0` here as retry is `0` initially, and I don't see where it gets a different value.

Reply:
It's a mistake, it should be `retry += 1`
When it reaches 5 it exits the loop.

Reply:
OK. clear!

Reply:
fixed in PR #187